var/spack/repos/builtin/packages/ncl/package.py

```diff
@@ -132,6 +132,7 @@ class Ncl(Package):
 
     sanity_check_is_file = ["bin/ncl"]
 
+    @run_before("install")
```

Comment:
This should happen immediately after the stage is created, `run_after` should not be needed. Can you check what is going on here?

Reply:
If everything works as expected now we are running this function 2 times.

Reply:
I think I see the issue now. The grib support changes from a prior PR introduced another `patch()` function that is triggered only when `+grib`. Since this `patch()` is defined second, it gets called and this other patch does not without the `run_before` decorator.

I'm thinking to submit a superseding PR to consolidate these two patches into one, with the `+grib` becoming a satisfies condition in the new patch. I can move the `filter_sbang` code into the patch as well, so that it gets applied when `spack patch` is called.

Reply:
That sounds good to me :metal: